src/dotnet/commands/dotnet-build/CompileContext.cs

```diff
@@ -342,37 +354,36 @@ private bool InvokeCompileOnRootProject()
 
             if (succeeded)
             {
-                MakeRunnableIfNecessary();
-            }            
-            
-            return succeeded;
-        }
-
-        private void MakeRunnableIfNecessary()
-        {
-            var compilationOptions = CompilerUtil.ResolveCompilationOptions(_rootProject, _args.ConfigValue);
-
-            // TODO: Make this opt in via another mechanism
-            var makeRunnable = compilationOptions.EmitEntryPoint.GetValueOrDefault() ||
-                               _rootProject.IsTestProject();
-
-            if (makeRunnable)
-            {
-                var outputPathCalculator = _rootProject.GetOutputPathCalculator(_args.OutputValue);
-                var rids = new List<string>();
-                if (string.IsNullOrEmpty(_args.RuntimeValue))
+                if (_rootProject.ProjectFile.HasRuntimeOutput(_args.ConfigValue))
                 {
-                    rids.AddRange(PlatformServices.Default.Runtime.GetAllCandidateRuntimeIdentifiers());
+                    MakeRunnable();
                 }
                 else
                 {
-                    rids.Add(_args.RuntimeValue);
+                    var calculator = _rootProject.GetOutputPathCalculator(_args.ConfigValue, _args.BuildBasePathValue, _args.OutputValue);
```

Comment:
You have the piece above wrapped in a method. Keep the level of abstraction in this method the name and wrap this block of code in a method as well.